### PR TITLE
ci: rename `material-unit-tests` job to `components-repo-unit-tests` job

### PR DESCRIPTION
**Pull request by devversion**
We rename the `material-unit-tests` job to `components-repo-unit-tests`
because the job runs all unit tests found in the Angular Components repository.
This includes the Angular CDK, Angular Material and more. Also the repository has
been renamed from `angular/material2` to `angular/components` in the past.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -27,10 +27,10 @@ var_4: &cache_key_fallback v3-angular-node-10.16-
 var_3_win: &cache_key_win v5-angular-win-node-12.0-{{ checksum "yarn.lock" }}-{{ checksum "WORKSPACE" }}-{{ checksum "packages/bazel/package.bzl" }}-{{ checksum "aio/yarn.lock" }}
 var_4_win: &cache_key_win_fallback v5-angular-win-node-12.0-
 
-# Cache key for the Material unit tests job. **Note** when updating the SHA in the cache keys,
-# also update the SHA for the "MATERIAL_REPO_COMMIT" environment variable.
-var_5: &material_unit_tests_cache_key v5-angular-material-97a7e2babbccd3dc58e7b3364004e45ed5bd9968
-var_6: &material_unit_tests_cache_key_fallback v5-angular-material-
+# Cache key for the `components_repo_unit_tests` job. **Note** when updating the SHA in the
```
Comment (gkalpak) [inline .circleci/config.yml]:
```suggestion
# Cache key for the `components-repo-unit-tests` job. **Note** when updating the SHA in the
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -27,10 +27,10 @@ var_4: &cache_key_fallback v3-angular-node-10.16-
...
+# cache keys also update the SHA for the "COMPONENTS_REPO_COMMIT" environment variable.
+var_5: &components_repo_unit_tests_cache_key v5-angular-components-97a7e2babbccd3dc58e7b3364004e45ed5bd9968
+var_6: &components_repo_unit_tests_cache_key_fallback v5-angular-components-
```
Comment (gkalpak) [inline .circleci/config.yml]: Given that `COMPONENT_REPO_TMP_DIR` has changed, I believe you should also update `v5` to `v6`.

**Review by gkalpak — CHANGES_REQUESTED** [review]
Thx!
<sub>This has been bothering me for a while :grin:</sub>

File: .circleci/config.yml (hunk 2 above)
Comment (devversion) [reply .circleci/config.yml]: I think we should be fine since the overall prefix of both cache key and fallback key have changed.

Comment (gkalpak) [reply .circleci/config.yml]: Ooops, right, I missed that 😇

**Comment by devversion** [comment]
@gkalpak Yeah 😄 This has been on my list for quite some time. Rebased and addressed your feedback!

**Review by gkalpak — APPROVED** [review]
✨

**Review by josephperrott — APPROVED** [review]
LGTM, global approval

**Comment by AndrewKushnir** [comment]
@devversion it looks like this PR now has a conflict with master branch (after merging another PR into master). Could you please rebase this PR and resolve the conflict? Thank you.

**Pull request closed without merging**